Detectors/TOF/simulation/include/TOFSimulation/Strip.h

```diff
@@ -0,0 +1,118 @@
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+//
+//  TOF strip class: it will be used to store the hits and digits at TOF that
+//  fall in the same strip
+//
+
+#ifndef ALICEO2_TOF_STRIP_H_
+#define ALICEO2_TOF_STRIP_H_
+
+#include <TOFBase/Digit.h>
+#include <TObject.h>
+#include <exception>
+#include <map>
+#include <sstream>
+#include <vector>
+#include "MathUtils/Cartesian3D.h"
+#include "TOFSimulation/Detector.h" // for HitType
+
+namespace o2
+{
+namespace tof
+{
+
+/// @class Strip
+/// @brief Container for similated points connected to a given TOF strip
+/// This will be used in order to allow a more efficient clusterization
+/// that can happen only between digits that belong to the same strip
+///
+
+class Strip
+{
+
+ public:
+  /// Default constructor
+  Strip() = default;
+
+  /// Destructor
+  ~Strip() = default;
+
+  /// Main constructor
+  /// @param stripindex Index of the strip
+  /// @param mat Transformation matrix
+  Strip(Int_t index);
+
+  /// Copy constructor
+  /// @param ref Reference for the copy
+  Strip(const Strip& ref);
+
+  /// Empties the point container
+  /// @param option unused
+  void clear();
+
+  /// Change the chip index
+  /// @param index New chip index
+  void setStripIndex(Int_t index) { mStripIndex = index; }
+  void init(Int_t index)
+  //, const o2::Transform3D* mat)
+  {
+    mStripIndex = index;
+    //mMat = mat;
+  }
+
+  /// Get the chip index
+  /// @return Index of the chip
+  Int_t getStripIndex() const { return mStripIndex; }
+  /// Insert new ITSMFT point into the Chip
+  /// @param p Hit to be added
+  void insertHit(const o2::tof::HitType* p);
+
+  /// Get the number of point assigned to the chip
+  /// @return Number of points assigned to the chip
+  Int_t getNumberOfHits() const { return mHits.size(); }
+  Int_t getNumberOfDigits() const { return mDigits.size(); }
+
+  /// Get the strip index from hit
+  Int_t getStripIndex(const o2::tof::HitType* hit);
+
+  /// reset points container
+  void clearHits() { mHits.clear(); }
+  o2::tof::Digit* findDigit(ULong64_t key);
+
+  /// Access Hit assigned to chip at a given index
+  /// @param index Index of the point
+  /// @return Hit at given index (nullptr if index is out of bounds)
+  const o2::tof::HitType* getHitAt(Int_t index) const;
+
+  Int_t addDigit(Double_t time, Int_t channel, Int_t tdc, Int_t tot, Int_t bc, Int_t lbl); // returns the MC label
+
+  void fillOutputContainer(std::vector<o2::tof::Digit>* digits);
+
+ protected:
+  Int_t mStripIndex = -1;                      ///< Strip ID
+  std::vector<const o2::tof::HitType*> mHits;  ///< Hits connected to the given strip
```

Comment:
Just in case (since it was copied from ITS): for the ITS I've abandoned this per-chip (strip in your case) buffer for hits, instead I just globally sort the hits according chip (strip) ID, then run the digitization looping over this sorted array. This increased both speed and memory performance.

Reply:
In our case, having the Strip intermediate structure seems for now ok, as we also order per time (needed for the clusterization) the hits/digits in the strip. We can anyway consider this in the next weeks.